PROJECT_MCPU/MCPU/DOC/GENERATED/man/man3/ServiceForm.3
.TH "ServiceForm" 3 "MCPU" \" -*- nroff -*-
.ad l
.nh
.SH NAME
ServiceForm
.SH SYNOPSIS
.br
.PP
.PP
\fR#include <ServiceForm\&.h>\fP
.PP
Inherits System::Windows::Forms::Form\&.
.SS "Public Types"

.in +1c
.ti -1c
.RI "enum class \fBpanels\fP { \fBPANEL_NOT_DEFINED\fP = 0, \fBMAIN_SERVICE_PANEL\fP, \fBCALIB_PANEL\fP, \fBSETUP_PANEL\fP, \fBCALIB_ZEROSETTING_PANEL\fP, \fBTOOL_EXPOSURE_PANEL\fP, \fBTOOL_ROTATION_PANEL\fP }"
.br
.in -1c
.SS "Public Member Functions"

.in +1c
.ti -1c
.RI "\fBServiceForm\fP (void)"
.br
.ti -1c
.RI "void \fBopen\fP (void)"
.br
.ti -1c
.RI "void \fBclose\fP (void)"
.br
.ti -1c
.RI "void \fBinitServiceStatus\fP (void)"
.br
.ti -1c
.RI "void \fBserviceStatusManagement\fP (void)"
.br
.ti -1c
.RI "void \fBsetActivePanel\fP (\fBServiceForm::panels\fP p)"
.br
.ti -1c
.RI "void \fBinitServicePanel\fP (void)"
.br
.ti -1c
.RI "void \fBcancServicePanel\fP (void)"
.br
.ti -1c
.RI "void \fBinitCalibrationPanel\fP (void)"
.br
.ti -1c
.RI "void \fBcancCalibrationPanel\fP (void)"
.br
.in -1c
.SS "Static Public Member Functions"

.in +1c
.ti -1c
.RI "static bool \fBisPasswordOperator\fP ()"
.br
.ti -1c
.RI "static bool \fBisPasswordFactory\fP ()"
.br
.in -1c
.SS "Public Attributes"

.in +1c
.ti -1c
.RI "System::Timers::Timer ^ \fBserviceTimer\fP"
.br
.ti -1c
.RI "bool \fBopen_status\fP"
.br
.ti -1c
.RI "HWND \fBwindow\fP"
.br
.ti -1c
.RI "\fBServiceForm::panels\fP \fBcurrent_panel\fP"
.br
.in -1c
.SS "Static Public Attributes"

.in +1c
.ti -1c
.RI "static \fBServiceForm\fP ^ \fBpanel\fP"
.br
.ti -1c
.RI "static bool \fBpasswd_operator\fP = false"
.br
.ti -1c
.RI "static bool \fBpasswd_factory\fP = false"
.br
.in -1c
.SS "Protected Member Functions"

.in +1c
.ti -1c
.RI "\fB~ServiceForm\fP ()"
.br
.RI "Clean up any resources being used\&. "
.ti -1c
.RI "virtual void \fBWndProc\fP (System::Windows::Forms::Message% m) override"
.br
.in -1c
.SS "Protected Attributes"

.in +1c
.ti -1c
.RI "Object ^ \fBpConfirmation\fP"
.br
.in -1c
.SS "Private Member Functions"

.in +1c
.ti -1c
.RI "void \fBformInitialization\fP (void)"
.br
.ti -1c
.RI "void \fBInitializeComponent\fP (void)"
.br
.RI "Required method for Designer support - do not modify the contents of this method with the code editor\&. "
.ti -1c
.RI "System::Void \fBonServiceTimeout\fP (Object^ source, System::Timers::ElapsedEventArgs^ e)"
.br
.ti -1c
.RI "System::Void \fBcancButton_Click\fP (System::Object^ sender, System::EventArgs^ e)"
.br
.ti -1c
.RI "System::Void \fBsystemCalibration_Click\fP (System::Object^ sender, System::EventArgs^ e)"
.br
.ti -1c
.RI "System::Void \fBzeroSetting_Click\fP (System::Object^ sender, System::EventArgs^ e)"
.br
.ti -1c
.RI "System::Void \fBexposureTool_Click\fP (System::Object^ sender, System::EventArgs^ e)"
.br
.ti -1c
.RI "System::Void \fBsystemRotation_Click\fP (System::Object^ sender, System::EventArgs^ e)"
.br
.ti -1c
.RI "System::Void \fBbiopsyTool_Click\fP (System::Object^ sender, System::EventArgs^ e)"
.br
.ti -1c
.RI "System::Void \fBbiopsyLightOn_Click\fP (System::Object^ sender, System::EventArgs^ e)"
.br
.ti -1c
.RI "System::Void \fBbiopsyLightOff_Click\fP (System::Object^ sender, System::EventArgs^ e)"
.br
.ti -1c
.RI "System::Void \fBbiopsyPark_Click\fP (System::Object^ sender, System::EventArgs^ e)"
.br
.in -1c
.SS "Private Attributes"

.in +1c
.ti -1c
.RI "System::Windows::Forms::PictureBox ^ \fBexposureTool\fP"
.br
.ti -1c
.RI "System::Windows::Forms::PictureBox ^ \fBbiopsyTool\fP"
.br
.ti -1c
.RI "System::Windows::Forms::PictureBox ^ \fBbiopsyLightOff\fP"
.br
.ti -1c
.RI "System::Windows::Forms::PictureBox ^ \fBbiopsyLightOn\fP"
.br
.ti -1c
.RI "System::Windows::Forms::PictureBox ^ \fBbiopsyPark\fP"
.br
.ti -1c
.RI "System::Windows::Forms::Label ^ \fBserviceMenuTitle\fP"
.br
.ti -1c
.RI "System::Windows::Forms::Label ^ \fBlabelDate\fP"
.br
.ti -1c
.RI "System::Windows::Forms::Label ^ \fBlabelInstallation\fP"
.br
.ti -1c
.RI "System::Windows::Forms::Label ^ \fBlabelTime\fP"
.br
.ti -1c
.RI "System::Windows::Forms::PictureBox ^ \fBserviceCanc\fP"
.br
.ti -1c
.RI "System::Windows::Forms::Panel ^ \fBservicePanel\fP"
.br
.ti -1c
.RI "System::Windows::Forms::PictureBox ^ \fBrotationTool\fP"
.br
.ti -1c
.RI "System::Windows::Forms::PictureBox ^ \fBsystemSetup\fP"
.br
.ti -1c
.RI "System::Windows::Forms::PictureBox ^ \fBsystemCalibration\fP"
.br
.ti -1c
.RI "System::Windows::Forms::Panel ^ \fBcalibPanel\fP"
.br
.ti -1c
.RI "System::Windows::Forms::PictureBox ^ \fBzeroSetting\fP"
.br
.ti -1c
.RI "System::ComponentModel::Container ^ \fBcomponents\fP"
.br
.RI "Required designer variable\&. "
.in -1c
.SH "Member Enumeration Documentation"
.PP 
.SS "enum class \fBServiceForm::panels\fP\fR [strong]\fP"

.PP
\fBEnumerator\fP
.in +1c
.TP
\f(BIPANEL_NOT_DEFINED \fP(0)
.TP
\f(BIMAIN_SERVICE_PANEL \fP
.TP
\f(BICALIB_PANEL \fP
.TP
\f(BISETUP_PANEL \fP
.TP
\f(BICALIB_ZEROSETTING_PANEL \fP
.TP
\f(BITOOL_EXPOSURE_PANEL \fP
.TP
\f(BITOOL_ROTATION_PANEL \fP
.SH "Constructor & Destructor Documentation"
.PP 
.SS "ServiceForm::ServiceForm (void )\fR [inline]\fP"

.SS "ServiceForm::~ServiceForm ()\fR [inline]\fP, \fR [protected]\fP"

.PP
Clean up any resources being used\&. 
.SH "Member Function Documentation"
.PP 
.SS "System::Void ServiceForm::biopsyLightOff_Click (System::Object^ sender, System::EventArgs^ e)\fR [private]\fP"

.SS "System::Void ServiceForm::biopsyLightOn_Click (System::Object^ sender, System::EventArgs^ e)\fR [private]\fP"

.SS "System::Void ServiceForm::biopsyPark_Click (System::Object^ sender, System::EventArgs^ e)\fR [private]\fP"

.SS "System::Void ServiceForm::biopsyTool_Click (System::Object^ sender, System::EventArgs^ e)\fR [private]\fP"

.SS "void ServiceForm::cancButton_Click (System::Object^ sender, System::EventArgs^ e)\fR [private]\fP"

.SS "void ServiceForm::cancCalibrationPanel (void )"

.SS "void ServiceForm::cancServicePanel (void )"

.SS "void ServiceForm::close (void )"

.SS "System::Void ServiceForm::exposureTool_Click (System::Object^ sender, System::EventArgs^ e)\fR [private]\fP"

.SS "void ServiceForm::formInitialization (void )\fR [private]\fP"

.SS "void ServiceForm::initCalibrationPanel (void )"

.SS "void ServiceForm::InitializeComponent (void )\fR [inline]\fP, \fR [private]\fP"

.PP
Required method for Designer support - do not modify the contents of this method with the code editor\&. 
.SS "void ServiceForm::initServicePanel (void )"

.SS "void ServiceForm::initServiceStatus (void )"

.SS "static bool ServiceForm::isPasswordFactory ()\fR [inline]\fP, \fR [static]\fP"

.SS "static bool ServiceForm::isPasswordOperator ()\fR [inline]\fP, \fR [static]\fP"

.SS "System::Void ServiceForm::onServiceTimeout (Object^ source, System::Timers::ElapsedEventArgs^ e)\fR [private]\fP"

.SS "void ServiceForm::open (void )"

.SS "void ServiceForm::serviceStatusManagement (void )"

.SS "void ServiceForm::setActivePanel (\fBServiceForm::panels\fP p)"

.SS "System::Void ServiceForm::systemCalibration_Click (System::Object^ sender, System::EventArgs^ e)\fR [private]\fP"

.SS "System::Void ServiceForm::systemRotation_Click (System::Object^ sender, System::EventArgs^ e)\fR [private]\fP"

.SS "void ServiceForm::WndProc (System::Windows::Forms::Message% m)\fR [override]\fP, \fR [protected]\fP, \fR [virtual]\fP"

.SS "System::Void ServiceForm::zeroSetting_Click (System::Object^ sender, System::EventArgs^ e)\fR [private]\fP"

.SH "Member Data Documentation"
.PP 
.SS "System::Windows::Forms::PictureBox ^ ServiceForm::biopsyLightOff\fR [private]\fP"

.SS "System::Windows::Forms::PictureBox ^ ServiceForm::biopsyLightOn\fR [private]\fP"

.SS "System::Windows::Forms::PictureBox ^ ServiceForm::biopsyPark\fR [private]\fP"

.SS "System::Windows::Forms::PictureBox ^ ServiceForm::biopsyTool\fR [private]\fP"

.SS "System::Windows::Forms::Panel ^ ServiceForm::calibPanel\fR [private]\fP"

.SS "System::ComponentModel::Container ^ ServiceForm::components\fR [private]\fP"

.PP
Required designer variable\&. 
.SS "\fBServiceForm::panels\fP ServiceForm::current_panel"

.SS "System::Windows::Forms::PictureBox ^ ServiceForm::exposureTool\fR [private]\fP"

.SS "System::Windows::Forms::Label ^ ServiceForm::labelDate\fR [private]\fP"

.SS "System::Windows::Forms::Label ^ ServiceForm::labelInstallation\fR [private]\fP"

.SS "System::Windows::Forms::Label ^ ServiceForm::labelTime\fR [private]\fP"

.SS "bool ServiceForm::open_status"

.SS "\fBServiceForm\fP ^ ServiceForm::panel\fR [static]\fP"

.SS "bool ServiceForm::passwd_factory = false\fR [static]\fP"

.SS "bool ServiceForm::passwd_operator = false\fR [static]\fP"

.SS "Object ^ ServiceForm::pConfirmation\fR [protected]\fP"

.SS "System::Windows::Forms::PictureBox ^ ServiceForm::rotationTool\fR [private]\fP"

.SS "System::Windows::Forms::PictureBox ^ ServiceForm::serviceCanc\fR [private]\fP"

.SS "System::Windows::Forms::Label ^ ServiceForm::serviceMenuTitle\fR [private]\fP"

.SS "System::Windows::Forms::Panel ^ ServiceForm::servicePanel\fR [private]\fP"

.SS "System::Timers::Timer ^ ServiceForm::serviceTimer"

.SS "System::Windows::Forms::PictureBox ^ ServiceForm::systemCalibration\fR [private]\fP"

.SS "System::Windows::Forms::PictureBox ^ ServiceForm::systemSetup\fR [private]\fP"

.SS "HWND ServiceForm::window"

.SS "System::Windows::Forms::PictureBox ^ ServiceForm::zeroSetting\fR [private]\fP"


.SH "Author"
.PP 
Generated automatically by Doxygen for MCPU from the source code\&.


PROJECT_MCPU/MCPU/MCPU_MASTER/WINDOWS/SERVICE/ServiceForm.cpp
#include "../AWS/awsProtocol.h"
#include "ServiceForm.h"
#include "ErrorForm.h"
#include "ConfigurationFiles.h"
#include "Notify.h"
#include "../gantry_global_status.h"

#include "ServiceExposureTool.h"
#include "ServiceRotationTool.h"
#include "ServiceZeroSettingTool.h"
#include "PCB325.h"


// Main Panel Definition
#define FORM_BACKGROUND Image::FromFile(Gantry::applicationResourcePath + "ServiceForm\\ServiceBackground.PNG")
#define SERVICE_MENU_BACKGROUND Image::FromFile(Gantry::applicationResourcePath + "ServiceForm\\ServiceMenuBackground.PNG")
#define CALIB_MENU_BACKGROUND System::Drawing::Image::FromFile(Gantry::applicationResourcePath + "ServiceForm\\CALIBRATION\\CalibrationBackground.PNG")


void ServiceForm::formInitialization(void) {

	// Initialize the position of the form
	this->Left = Gantry::monitor_X0;
	this->Top = Gantry::monitor_Y0;

	// Common Items
	BackgroundImage = FORM_BACKGROUND;
	serviceMenuTitle->Text = "";
	labelInstallation->Text = SystemConfig::Configuration->getParam(SystemConfig::PARAM_INSTALLATION_NAME)[SystemConfig::PARAM_INSTALLATION_NAME_TOP];
	serviceCanc->BackColor = Color::Transparent;
	
	servicePanel->SetBounds(0, 0, PANEL_WIDTH, PANEL_HIGH);
	servicePanel->BackgroundImage = SERVICE_MENU_BACKGROUND;
	servicePanel->Location = System::Drawing::Point(PANEL_X, PANEL_Y);
	systemCalibration->BackColor = Color::Transparent;
	systemSetup->BackColor = Color::Transparent;
	rotationTool->BackColor = Color::Transparent;

	// Calib panel Definition
	calibPanel->SetBounds(0, 0, PANEL_WIDTH, PANEL_HIGH);
	calibPanel->BackgroundImage = CALIB_MENU_BACKGROUND;
	calibPanel->Location = System::Drawing::Point(PANEL_X, PANEL_Y);

	// Set of the Menu Buttons
	zeroSetting->BackColor = Color::Transparent;
	exposureTool->BackColor = Color::Transparent;
	biopsyTool->BackColor = Color::Transparent;
	biopsyLightOn->BackColor = Color::Transparent;
	biopsyLightOff->BackColor = Color::Transparent;
	biopsyPark->BackColor = Color::Transparent;

	serviceTimer = gcnew System::Timers::Timer(100);
	serviceTimer->Elapsed += gcnew System::Timers::ElapsedEventHandler(this, &ServiceForm::onServiceTimeout);
	serviceTimer->Stop();

	this->Hide();
	open_status = false;
}

void ServiceForm::initServiceStatus(void) {
	System::DateTime date;
	date = System::DateTime::Now;

	
	current_panel = panels::PANEL_NOT_DEFINED;
	setActivePanel(panels::MAIN_SERVICE_PANEL);

	// Activate the Operating Status  manual modes
	Gantry::setManualRotationMode(Gantry::manual_rotation_options::GANTRY_MANUAL_ROTATION_DISABLED);

	
	// Start the startup session	
	serviceTimer->Start();	
	
}

void ServiceForm::open(void) {
	if (open_status) return;
	open_status = true;	
	initServiceStatus();

	this->Show();
}

void ServiceForm::close(void) {
	if (!open_status) return;
	open_status = false;
	serviceTimer->Stop();
	this->Hide();
}

void ServiceForm::setActivePanel(ServiceForm::panels p) {
	if (p == current_panel) return;
	current_panel = p;

	// Hides all the panels
	servicePanel->Hide();
	calibPanel->Hide();
	
	

	// Stops the timer: each individual panel starts its timer version 
	serviceTimer->Stop();

	switch (p) {
	case panels::MAIN_SERVICE_PANEL:
		initServicePanel();
		servicePanel->Show();
		break;
	case panels::CALIB_PANEL:
		initCalibrationPanel();
		calibPanel->Show();
		break;

	
	}
}

void ServiceForm::serviceStatusManagement(void) {

	// Update the date time fields
	System::DateTime date;
	date = System::DateTime::Now;

	labelDate->Text = date.Day + ":" + date.Month + ":" + date.Year;
	labelTime->Text = date.Hour + ":" + date.Minute + ":" + date.Second;

}

void ServiceForm::WndProc(System::Windows::Forms::Message% m)
{
	switch (m.Msg) {

	case (WM_USER + 1): // on statusTimer Timeout
		
		serviceStatusManagement();
		break;
	}

	Form::WndProc(m);
}


System::Void ServiceForm::onServiceTimeout(Object^ source, System::Timers::ElapsedEventArgs^ e)
{
	
	SendMessageA(window, WM_USER + 1, 0, 0);
}



void ServiceForm::initServicePanel(void) {
	serviceMenuTitle->Text = Notify::TranslateLabel(Notify::messages::LABEL_SERVICE_PANEL_TITLE);

	return;
}
void ServiceForm::cancServicePanel(void) {	
	Gantry::setIdle();
	return;
}

System::Void  ServiceForm::systemCalibration_Click(System::Object^ sender, System::EventArgs^ e) {
	setActivePanel(panels::CALIB_PANEL);
}




// Canc button common management
void ServiceForm::cancButton_Click(System::Object^ sender, System::EventArgs^ e) {
	switch (current_panel) {
		case panels::MAIN_SERVICE_PANEL: cancServicePanel(); break;
		case panels::CALIB_PANEL: cancCalibrationPanel(); break;
		
		

	}

}

void ServiceForm::cancCalibrationPanel(void) {
	setActivePanel(panels::MAIN_SERVICE_PANEL);
	return;
}

void ServiceForm::initCalibrationPanel(void) {
	serviceMenuTitle->Text = Notify::TranslateLabel(Notify::messages::LABEL_CALIBRATION_PANEL_TITLE);
	return;
}



System::Void ServiceForm::exposureTool_Click(System::Object^ sender, System::EventArgs^ e) {
	ServiceExposureTool::panel->dialog_open(this); // Opens as a Dialog the service Panel
}
System::Void  ServiceForm::systemRotation_Click(System::Object^ sender, System::EventArgs^ e) {
	ServiceRotationTool::panel->dialog_open(this); // Opens as a Dialog the service Panel
}
System::Void ServiceForm::zeroSetting_Click(System::Object^ sender, System::EventArgs^ e) {
	ServiceZeroSettingTool::panel->dialog_open(this); // Opens as a Dialog the service Panel
}

System::Void ServiceForm::biopsyTool_Click(System::Object^ sender, System::EventArgs^ e) {
	PCB325::activateCycleTest();
}

System::Void ServiceForm::biopsyLightOn_Click(System::Object^ sender, System::EventArgs^ e) {
	
	try {
		System::Byte ena = System::Convert::ToByte(BiopsyConfig::Configuration->getParam(BiopsyConfig::PARAM_POWER_LIGHT)[BiopsyConfig::PARAM_POWER_LIGHT_ENABLE]);
		int duty = System::Convert::ToInt16(BiopsyConfig::Configuration->getParam(BiopsyConfig::PARAM_POWER_LIGHT)[BiopsyConfig::PARAM_POWER_LIGHT_DUTY]);
		PCB325::setLight(ena, duty);
	}
	catch (...) {
		PCB325::setLight(false, 0);
	}
	
}
System::Void ServiceForm::biopsyLightOff_Click(System::Object^ sender, System::EventArgs^ e) {
	PCB325::setLight(false, 0);
}

System::Void ServiceForm::biopsyPark_Click(System::Object^ sender, System::EventArgs^ e) {
	PCB325::moveHome();
	return;
}
